### PR TITLE
Spark Max Implementation

### PR DESCRIPTION
**Pull request by theVerySharpFlat**
This is the first edition of the SparkMax velocity controller abstraction. I checked the math in the conversions and it seems to work out alright assuming the multiplication in getVelocity() works out. However, it would be great if somebody could give it a read-through. 

A little on the SparkMaxes: Units are done in RPM. I chose the base units of RawUnit to be radians as it is a pretty easy conversion 1 rotation * 2pi = 2pi rad.

### CODE REVIEW

**Comment by theVerySharpFlat** [comment]
There are some things that the Talon implementation has but this doesn't.

 I could not make ```rmb::SparkMaxVelocityController``` inherit from ```frc::MotorSafety``` as ``` rev::CANSparkMax``` does not inherit from ```frc::MotorSafety``` like ```ctre::phoenix::motorcontrol::can::WPI_TalonSRX``` does. I really don't think we should go about implementing the functions ourself. 

Also, the function for setting closed loop error on the SparkMaxes is [SparkMaxPIDController::SetSmartMotionAllowedClosedLoopError](https://codedocs.revrobotics.com/cpp/classrev_1_1_spark_max_p_i_d_controller.html#a97bb4fa3723e68485d8ed55f09ad56e6) which only works in SmartMotion mode. I haven't really looked into smart motion, but is it something we are going to use?

**Comment by theVerySharpFlat** [comment]
also, could somebody please help me format this constructor?

https://github.com/rambunction4330/RobotCode2022/blob/80310ec01c86a0c22368f8c367706c9169a8fd1f/lib/src/motorcontrol/SparkMax/SparkMaxVelocityController.cpp#L14-L23

it looks disgusting and I don't know how to fix it.

File: lib/include/rmb/motorcontrol/SparkMax/SparkMaxVelocityController.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    // user defined conversion factor
+    using ConversionUnit   = typename units::compound_unit<DistanceUnit, units::inverse<units::radians>>;
+    using ConversionUnit_t = typename units::unit_t<ConversionUnit>;
+
+    // raw velocity is in rpm. 1 rotation * 2pi rad = 2pi rad
+    using RawUnit   = typename units::unit<std::ratio<2>, units::radians, std::ratio<1>>;
+    using RawUnit_t = typename units::unit_t<RawUnit>;
+    using RawVelocity   = typename units::compound_unit<RawUnit, units::inverse<units::seconds>>;
+    using RawVelocity_t = typename units::unit_t<RawVelocity>;
```
Comment (gjurgiel) [inline lib/include/rmb/motorcontrol/SparkMax/SparkMaxVelocityController.h]: The raw velocity should be in rotations per minute not second, right?

**Comment by gjurgiel** [comment]
> There are some things that the Talon implementation has but this doesn't.
> 
> I could not make `rmb::SparkMaxVelocityController` inherit from `frc::MotorSafety` as ` rev::CANSparkMax` does not inherit from `frc::MotorSafety` like `ctre::phoenix::motorcontrol::can::WPI_TalonSRX` does. I really don't think we should go about implementing the functions ourself.
> 
> Also, the function for setting closed loop error on the SparkMaxes is [SparkMaxPIDController::SetSmartMotionAllowedClosedLoopError](https://codedocs.revrobotics.com/cpp/classrev_1_1_spark_max_p_i_d_controller.html#a97bb4fa3723e68485d8ed55f09ad56e6) which only works in SmartMotion mode. I haven't really looked into smart motion, but is it something we are going to use?

- I'll go back and remove motor safety from the Talons because it Is definitely unnecessary. 
- If the SparkMax doesn't have a function for allowable velocity error, jut don't implement it. 
- Smart motion is the control mode we will use for the Position controller, that does both PID and motion profiling. The SparkMax has a similar control mode for. velocity call kSmartVelcoity that might do a similar thing, but the docs are a little unclear on what exactly it does.

File: lib/include/rmb/motorcontrol/SparkMax/SparkMaxVelocityController.h (hunk 1 above)
Comment (theVerySharpFlat) [reply lib/include/rmb/motorcontrol/SparkMax/SparkMaxVelocityController.h]: That is correct. I will fix at that meeting. Thanks for catching that.

**Pull request merged**